### ISSUE TITLE
Enable branch protection rules

### ISSUE THREAD

**Issue by ferrarimarco** [github/super-linter#991]
After having a look at the Git history, which is a bit hard to follow, I think we might benefit from a couple more branch protection rules for the main/master branch.

1. Require linear history.
1. Require branches to be up to date before merging (also contributing to #982)

Example of the current history:

<img width="402" alt="Screenshot 2020-11-12 at 17 23 22" src="https://user-images.githubusercontent.com/621939/98966703-deeeec00-250b-11eb-81d7-23fb1dbf827c.png">

@admiralAwkbar Thoughts?

**Comment by admiralAwkbar** [comment]
@ferrarimarco I love it.. we need it... consider it done

**Comment by admiralAwkbar** [comment]
@ferrarimarco updated and done!
thank you for keeping up with this!

**Comment by ferrarimarco** [comment]
Re-opening to bug more :)

**Comment by ferrarimarco** [comment]
Forgot to ask about "Squash and Merge" to be the only merging mechanism, otherwise we end up with the same result as before

**Comment by admiralAwkbar** [comment]
@ferrarimarco The only option I had was squash and merge
Is that what you were looking for?

**Comment by ferrarimarco** [comment]
Yes! that one :) nice!
